client/client-lib/src/query.rs

```diff
@@ -9,16 +9,17 @@ use jsonrpsee_types::error::CallError as RpcCallError;
 use threshold_crypto::PublicKey;
 use tracing::debug;
 
-use crate::api::{FedResponse, Result};
-use crate::ApiError;
+use crate::{api, ApiError};
 
 /// Returns a result from the first responding peer
 pub struct TrustAllPeers;
 
 impl<R> QueryStrategy<R> for TrustAllPeers {
-    fn process(&mut self, response: FedResponse<R>) -> QueryStep<R> {
-        let FedResponse { peer: _, result } = response;
-        QueryStep::Finished(result.map_err(ApiError::RpcError))
+    fn process(&mut self, peer: PeerId, result: api::Result<R>) -> QueryStep<R> {
+        match result {
+            Ok(o) => QueryStep::Success(o),
```

Comment:
@jkitman I changed this a little. I don't think we want to return a random failure, just because it was the first one to arrive. Also query strategy returns a `FedResponse` which is a collection of errors, not just one error. So we keep waiting until someone returns a success or everyone return a failure.

Reply:
My thinking was that if enough peers agree on the same error then we can trust that is the "correct" error, e.g. a 404 indicates a tx id was not found.  Although perhaps we should avoid treating errors as informative for the client.